docs/Configuration.md

```diff
@@ -1,107 +1,218 @@
+<!-- DO NOT MODIFY THIS FILE DIRECTLY, IT IS AUTO-GENERATED BY [org.apache.gluten.config.AllGlutenConfiguration] -->
+
 ---
+
 layout: page
 title: Configuration
 nav_order: 3
----
+------------
 
 # Spark Configurations for Gluten Plugin
 
 There are many configurations could impact the Gluten Plugin performance and can be fine-tuned in Spark.
 You can add these configurations into spark-defaults.conf to enable or disable the setting.
 
-## Spark parameters
-
-| Parameters                                                   | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | Recommend Setting                                    |
-|--------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------|
-| spark.driver.extraClassPath                                  | Gluten Plugin jar file to prepend to the classpath of the driver.                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | /path/to/jar_file                                    |
```

Comment:
We need something to do before regenrating the `Configuration.md`.
1. A way to define the spark parameters that used for gluten;
2. Synchronize the configuration information in `Configuration.md` and `GlutenConfig`, because some configurations are not well defined in GlutenConfig;
3. Use a unified interface to define the gluten configs, such as internal, backends, and so on, so we can define how to show each config in `Configuration.md`

Reply:
There is a issue #6970 that discussed `GlutenConfig` before.